main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidator.java

```diff
@@ -41,27 +42,35 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    final Map<String, GtfsFareRule> fareRuleByZoneIdFieldsCombination =
+    final Map<Integer, GtfsFareRule> fareRuleByZoneIdFieldsCombination =
         new HashMap<>(fareRuleTable.entityCount());
     fareRuleTable
         .getEntities()
         .forEach(
             fareRule -> {
-              String fieldsCombination =
-                  fareRule.routeId()
-                      + fareRule.originId()
-                      + fareRule.containsId()
-                      + fareRule.destinationId();
-              if (fareRuleByZoneIdFieldsCombination.containsKey(fieldsCombination)) {
+              int fieldsCombinationHash = getZoneFieldsCombinationHash(fareRule);
+              if (fareRuleByZoneIdFieldsCombination.containsKey(fieldsCombinationHash)) {
                 noticeContainer.addValidationNotice(
                     new DuplicateFareRuleZoneIdFieldsNotice(
                         fareRule.csvRowNumber(),
                         fareRule.fareId(),
-                        fareRuleByZoneIdFieldsCombination.get(fieldsCombination).csvRowNumber(),
-                        fareRuleByZoneIdFieldsCombination.get(fieldsCombination).fareId()));
+                        fareRuleByZoneIdFieldsCombination.get(fieldsCombinationHash).csvRowNumber(),
+                        fareRuleByZoneIdFieldsCombination.get(fieldsCombinationHash).fareId()));
               } else {
-                fareRuleByZoneIdFieldsCombination.put(fieldsCombination, fareRule);
+                fareRuleByZoneIdFieldsCombination.put(fieldsCombinationHash, fareRule);
               }
             });
   }
+
+  /**
+   * Returns the hashcode associated to the combination of `fare_rules.origin_id`,
+   * `fare_rules.contains_id` and `fare_rules.destination_id`.
+   *
+   * @param fareRule
+   * @return the hashcode associated to the combination of this {@code GtfsFareRule} `origin_id`,
+   *     `contains_id` and `destination_id`.
+   */
+  private int getZoneFieldsCombinationHash(GtfsFareRule fareRule) {
```

Comment:
For code readability I think you could shorten this method name to just `getHash()`. The Javadocs and code give an explanation of what it includes.